WorkflowUI/Sources/ViewControllerDescription/ViewControllerDescription.swift

```diff
@@ -118,12 +125,50 @@
                 """
                 `ViewControllerDescription` was provided a view controller it cannot update: (\(viewController).
 
-                The view controller type (\(type(of: viewController)) is a compatible type to the expected type \(viewControllerType)).
+                The view controller type (\(type(of: viewController)) is a compatible type to the expected type \(kind.viewControllerType)).
                 """
             )
 
             update(viewController)
         }
     }
 
+    extension ViewControllerDescription {
+        /// Describes the `UIViewController` type that backs the `ViewControllerDescription`
+        /// in a way that is `Equatable` and `Hashable`. When implementing view controller
+        /// updating and diffing, you can use this type to identify if the backing view controller
+        /// type changed.
+        public struct KindIdentifier: Hashable {
```

Comment:
Since this type is `Hashable`, you can use it to salt your ID types, making handling diffs easier when the underlying type of the VC in the view description changes.